### PR TITLE
Feat: Add Session storage to store data on client storage

### PR DESCRIPTION
**Pull request by TG199**
### Type of change

- [ ] New feature (non-breaking change which adds functionality)

### Description of change
Implemented SessionStorage class for managing session-related data in sessionStorage, enhancing project functionality

### CODE REVIEW

**Comment by TG199** [comment]
Why are they failing 😥

File: reflex/__init__.py [hunk 1]
```diff
@@ -301,7 +301,7 @@
         "var",
         "Cookie",
         "LocalStorage",
-        "ComponentState",
+        "SessionStorage" "ComponentState",
```
Comment (masenf) [inline reflex/__init__.py:304]:
```suggestion
        "SessionStorage", "ComponentState",
```
```suggestion
        "SessionStorage" "ComponentState",
```

File: reflex/state.py [hunk 2]
```diff
@@ -2835,6 +2835,54 @@ def __new__(
...
+        """
+        if encoding or errors:
+            inst = super().__new__(cls, object, encoding or "utf-8", errors or "strict")
+        else:
+            inst = super().__new__(cls, object)
+        inst.name = name
+        inst.sync = sync
+        inst.session_id = session_id
+        inst.auth = auth
+        inst.timeout = timeout
```
Comment (masenf) [inline reflex/state.py:2882]: how do these get handled on the frontend / where did they come from?

**Review by masenf — CHANGES_REQUESTED** [review]
there should be more line diff in `state.js`, basically `hydrateClientStorage` and `applyClientStorageDelta` should be updated to handle the session storage cases as well

File: reflex/.templates/web/utils/state.js [hunk 3]
```diff
@@ -506,7 +518,18 @@ export const hydrateClientStorage = (client_storage) => {
       }
     }
   }
-  if (client_storage.cookies || client_storage.local_storage) {
+  if (client_storage.sessionStorage && typeof window != "undefined") {
```
Comment (masenf) [inline reflex/.templates/web/utils/state.js:521]:
```suggestion
  if (client_storage.session_storage && typeof window != "undefined") {
```

I think the key is `session_storage`, here and in various places below,

**Review by masenf — CHANGES_REQUESTED** [review]
how did you test this? I tried extending `integration/test_client_storage.py` to add some `SessionStorage` state vars and they didn't get saved, so something isn't quite working.

I'll push uo the tests that i wrote quickly so you can run them, but its not necessarily exhaustive.

**Comment by TG199** [comment]
Finally!

**Review by masenf — APPROVED** [review]
Looks good!

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -31,4 +31,4 @@ repos:
         always_run: true
         language: system
         description: 'Update pyi files as needed'
-        entry: python scripts/make_pyi.py
+        entry: python3 scripts/make_pyi.py
```
Comment (picklelo) [inline .pre-commit-config.yaml:34]: do we need this change?

File: reflex/event.py [hunk 5]
```diff
@@ -612,6 +612,34 @@ def remove_local_storage(key: str) -> EventSpec:
     )
 
 
+def clear_session_storage() -> EventSpec:
+    """Set a value in the session storage on the frontend.
```
Comment (picklelo) [inline reflex/event.py:616]: should be "clear all values", looks like we missed this in the local_storage also

File: reflex/event.py [hunk 6]
```diff
@@ -612,6 +612,34 @@ def remove_local_storage(key: str) -> EventSpec:
...
+        EventSpec: An event to clear the session storage.
+    """
+    return server_side(
+        "_clear_session_storage",
+        get_fn_signature(clear_session_storage),
+    )
+
+
+def remove_session_storage(key: str) -> EventSpec:
+    """Set a value in the session storage on the frontend.
```
Comment (picklelo) [inline reflex/event.py:628]: Should be "remove a value"

**Review by picklelo — APPROVED** [review]
Some minor docstring issues that we can fix in a follow up

**Pull request merged**